src/unix.rs

```diff
@@ -46,6 +40,21 @@ pub fn terminal_size_using_fd(fd: RawFd) -> Option<(Width, Height)> {
     }
 }
 
+/// Returns the size of the terminal using the given raw file descriptor, if available.
+///
+/// The given file descriptor must be an open file descriptor.
+///
+/// If the given file descriptor is not a tty, returns `None`
+#[deprecated(note = "Use `terminal_size_of` instead.
+     Use `BorrowedFd::borrow_raw` to convert a raw fd into a `BorrowedFd` if needed.")]
+pub fn terminal_size_using_fd(fd: RawFd) -> Option<(Width, Height)> {
+    // SAFETY: Under I/O safety, this function should be `unsafe`, but we can't
+    // remove it without breaking compatibility, so we instead deprecate it.
+    // This unsafe block has the same precondition that the function implicitly
+    // does: `fd` must be an open handle.
+    unsafe { terminal_size_of(BorrowedFd::borrow_raw(fd)) }
+}
```

Comment:
I would be OK making this function `unsafe` and bumping terminal-size's semver from 0.3 to 0.4.

Reply:
I've now added a patch making it `unsafe`.

Reply:
Presumably we want the same change applied to the windows version, too, right?

Reply:
Oops, yes. Fixed.